src/main/java/calculator/StringCalculator.kt

```diff
@@ -0,0 +1,14 @@
+package calculator
+
+import calculator.extractor.TextDelimiterExtractorFactory
+
+data class StringCalculator(
+    private val factory: TextDelimiterExtractorFactory,
+) {
+
+    fun add(text: String?): Int {
```

Comment:
애초에 메서드 파라미터를 `non-null` 타입으로 둔다면 불필요한 엘비스 연산자처리는 제거할 수 있을 것 같아요. 😃
굳이 필요하지 않다면 타입으로 먼저 제한해버리는게 추후 null 처리를 덜 해줄 수 있어 좋다고 생각해요. 

```suggestion
    fun add(text: String): Int {
```

Reply:
아.. 이게 조금 애매한게 1단계 - 문자열 덧셈 계산기 페이지에서 제공하는 기본 테스트 코드 중 `@NullAndEmptySource`로 테스트 값을 전달하는 케이스가 있어서 불가피하게 메서드 파라미터를 nullable하게 두었습니다 ㅠ

보통은 저도 불필요한 엘비스 연산자 처리를 지양하고 있어서 최대한 not-null 타입으로 코딩하려고 하고 있습니다 ㅎㅎ 과제에서 테스트 코드를 제공하고 있어서 수정없이 하고자 일단 다음처럼 구현했어요!

Reply:
아.. 생각을 못했네요. 
미션의 테스트 케이스에서 입력으로 null이 들어온 경우에 대한 예외 테스트도있었다는걸 깜빡했습니다. 😃 
(테스트 케이스를 커버할 수 있도록 지금처럼 메서드 파라미터 타입을 nullable하게 두시면 될 것 같습니다. 🙏)
남겨주신 코멘트를보니 이미 nullable 타입을 지양하고 계신듯하네요. 😃